gitlab/resource_gitlab_project_access_token.go

```diff
@@ -0,0 +1,180 @@
+package gitlab
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+func resourceGitlabProjectAccessToken() *schema.Resource {
+	// lintignore: XR002 // TODO: Resolve this tfproviderlint issue
+	return &schema.Resource{
+		Create: resourceGitlabProjectAccessTokenCreate,
+		Read:   resourceGitlabProjectAccessTokenRead,
+		Delete: resourceGitlabProjectAccessTokenDelete,
+
+		Schema: map[string]*schema.Schema{
+			"project": {
+				Type:     schema.TypeInt,
+				Required: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"scopes": {
+				Type:     schema.TypeSet,
+				Required: true,
+				ForceNew: true,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validation.StringInSlice([]string{"api", "read_api", "read_repository", "write_repository"}, false),
+				},
+			},
+			"expires_at": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: func(i interface{}, k string) (warnings []string, errors []error) {
+					v, ok := i.(string)
+					if !ok {
+						errors = append(errors, fmt.Errorf("expected type of %q to be string", k))
+						return warnings, errors
+					}
```

Comment:
Nit-pick: This string-type validation should not be necessary since you've already specified the type in the schema, and the docs say Terraform guarantees that `i` is "guaranteed to be of the proper Schema type".
```suggestion
					v := i.(string)
```

Reply:
Done